File: SecondPhaseGate(3in1)Java/MenstrualAppPrototype.java
import java.util.Scanner;
public static void main(String[] args) {
    MenstrualApp menstrualApp = new MenstrualApp();
    Scanner userInputCollection = new Scanner(System.in);

    int cycleLength;
    int periodLength;

    while (true) {
        try {
            System.out.print("\nEnter your last menstrual period date (yyyy-MM-dd): ");
            String lastDate = userInputCollection.nextLine();
            menstrualApp.lastDate(lastDate);
            break;
        } catch (Exception e) {
            System.out.println("\nInvalid date entered. Please try again.");
        }
    }

    while(true) {
        try {
            System.out.print("\nEnter your main cycle length: ");
            String cycleLengthStr = userInputCollection.nextLine();
            if(!cycleLengthStr.matches("\\d+")) {
                System.out.print("\nInvalid input. Please try again.: ");
            }else {
                cycleLength = Integer.parseInt(cycleLengthStr);
                menstrualApp.cycleLength(cycleLength);
                break;
            }
        }catch(Exception e) {
            System.out.println(e.getMessage());
        }
    }

    while(true) {
        try {
            System.out.print("\nEnter your main period length: ");
            String periodLengthStr = userInputCollection.nextLine();
            if(!periodLengthStr.matches("\\d+")) {
                System.out.print("\nInvalid input. Please try again.: ");
            }else {
                periodLength = Integer.parseInt(periodLengthStr);
                menstrualApp.periodLength(periodLength);
                break;
            }
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }

    System.out.println("\nYour next period date is: " + menstrualApp.predictNextPeriodDate());
    System.out.println("Your ovulation date is: " + menstrualApp.calculateOvulationDate());
    System.out.println("Your fertile start date is: " + menstrualApp.calculateFertileStartDate());
    System.out.println("Your fertile end date is: " + menstrualApp.calculateFertileEndDate());
    System.out.println("Stay safe ma'am🤣");
}

File: SecondPhaseGate(3in1)javaScript/MenstrualAppPrototype.js
const MenstrualApp = require('./MenstrualApp.js');
const menstrualApp = new MenstrualApp();
console.log("\nWelcome to your menstrual cycle tracker app");

const prompt = require('prompt-sync')();

while (true) {
    let choice = prompt("\nEnter your last menstrual period date (yyyy-MM-dd): ");
    try {
        menstrualApp.lastDate(choice);
        break;
    } catch (error) {
        console.error(error.message);
    }
}

while (true) {
    let cycleLength = prompt("\nEnter your cycle length in days: ");
    if (!isNaN(cycleLength) && cycleLength > 0) {
        menstrualApp.cycleLength(cycleLength);
        break;
    } else {
        console.error("\nInvalid cycle length. Please enter a positive number.");
    }
}

while (true) {
    let periodLength = prompt("\nEnter your period length in days: ");
    if (!isNaN(periodLength) && periodLength > 0 && periodLength <= menstrualApp.mainCycleLength) {
        menstrualApp.periodLength(periodLength);
        break;
    } else {
        console.error("Invalid period length. Please enter a positive number that is less than or equal to your cycle length.");
    }
}

console.log("Your next period date is: " + menstrualApp.nextPeriodDate());
console.log("Your ovulation date is: " + menstrualApp.ovulationDate());
console.log("Your fertile start date is: " + menstrualApp.fertileStartDate());
console.log("Your fertile end date is: " + menstrualApp.fertileEndDate());
console.log("Thank you for using the menstrual cycle tracker app. Have a great day!");